kong/llm/plugin/base.lua

```diff
@@ -104,8 +104,11 @@ function MetaPlugin:header_filter(sub_plugin, conf)
   if kong.response.get_source() == "service" then
     -- we use openai's streaming mode (SSE)
     if get_global_ctx("stream_mode") then
-      -- we are going to send plaintext event-stream frames for ALL models
+      -- we are going to send plaintext event-stream frames for ALL models,
+      -- but we capture the original incoming content-type for the chunk-parser later.
+      set_global_ctx("stream_content_type", kong.service.response.get_header("Content-Type"))
```

Comment:
let's move this to parse-request.lua

Reply:
Nope, here we need to store the upstream response `Content-Type` header because it will reset by the next line.

Reply:
`parse_request` does not fit this situation.